ext/mysqlnd/mysqlnd_ps.c

```diff
@@ -92,7 +92,7 @@ MYSQLND_METHOD(mysqlnd_stmt, store_result)(MYSQLND_STMT * const s)
 
 	/* Nothing to store for UPSERT/LOAD DATA*/
 	if (!mysqlnd_stmt_check_state(stmt)) {
-		SET_CLIENT_ERROR(conn->error_info, CR_COMMANDS_OUT_OF_SYNC, UNKNOWN_SQLSTATE, mysqlnd_out_of_sync);
+		SET_CLIENT_ERROR(stmt->error_info, CR_COMMANDS_OUT_OF_SYNC, UNKNOWN_SQLSTATE, mysqlnd_out_of_sync);
```

Comment:
Something that's not really clear to me is when we should be setting the error on the stmt only, and when we should be setting it on stmt and conn both. It seems like in this code we often set both, but it's not really consistent.

Reply:
These errors are an absolute mess. It should be that when a function starts with `mysql_stmt_` we set `stmt->error_info` only. For everything else we set `conn->error_info`. There is a problem with out of memory errors. There are also some exceptions to this rule e.g. `prepare` method which needs to set both errors as it can be called in both ways.

Reply:
Thanks, that makes sense.